gulpfile.js

```diff
@@ -1,127 +1,124 @@
-gulp.task('watch', gulp.parallel(tasks.watch));
-tasks.default.push('watch');
-gulp.task('default', gulp.series(
-  'compile',
-  gulp.parallel(tasks.default)
-));
-
-
-
-// Hold a reference to Webpack Dev Server when it is created
-let wpds = null;
-
-// Trigger a full reload of the Webpack Dev Server page
-function reloadWebpackDevServer(cb) {
-  // Bail if there is now WPDS for some reason
-  if (wpds === null) {
-    cb();
-    return false;
-  }
-
-  console.log('Reload Webpack Dev Server!');
-
-  // Nukes the "state" hash
-  wpds.sockWrite(wpds.sockets, 'hash', "");
-  // Then sends the signal that compares the previous hash, causing a FULL refresh
-  wpds.sockWrite(wpds.sockets, 'ok');
-  cb();
-  return true;
-}
+// Load up the function that will be used to start a webpack dev server
+// This does NOT start the server, that requires the gulp task below.
+const webpackdevserver = require('./tools/tasks/webpack-dev-server')(wpconfig, localhost, serverconfig);
 
 /**
- * Starts up the Webpack Dev Server and does the config adjustments that this
- * command does:
- *   webpack-dev-server --hot --inline --progress
- *
- * Note the absolute requirements for Hot Module Reloading in this Dev Server:
- *  - HtModuleReplacement plugin
- *  - Added entry points
+ * Starts up the Webpack Dev Server with our config from aove
  */
-gulp.task('webpack:server', (cb) => {
-
-  // Add HotModuleReplacementPlugin to the end of the webpack plugins
-  wpconfig.plugins.push(new webpack.HotModuleReplacementPlugin());
-  // Set these new entry points required for Hot Module replacement
-  wpconfig.entry['pattern-lab'].unshift(
-    `webpack-dev-server/client?${localhost}/`,
-    'webpack/hot/dev-server',
-  );
-
-  // Make a new server and store a reference to it so we can interact with it later
-  wpds = new WebpackDevServer(webpack(wpconfig), {
-    // ie http://localhost:8080/temp
-    publicPath: `${localhost}${wpconfig.output.publicPath}`,
-    // ie pattern-lab/public
-    contentBase: path.resolve(__dirname, 'pattern-lab', 'public'),
-    hot: true,
-    historyApiFallback: true,
-    inline: true,
-    stats: {
-      colors: true,
-    }
-  });
+gulp.task('webpack:server', webpackdevserver);
 
-  wpds.listen(8080, 'localhost', function (err, result) {
-    if (err) {
-      cb(err);
-      return false;
-    }
 
-    console.log(`Listening at ${localhost}`);
-    cb();
-    return true;
-  });
-});
+/**
+ * Sass-to-JSON
+ */
+const scssConfigs = [
+  {
+    src: './source/_patterns/00-base/05-colors/_colors.scss',
+    dest: './source/_patterns/00-base/05-colors/colors.json',
+    lineStartsWith: '$c-',
+    allowVarValues: false,
+  },
+  {
+    src: './source/_patterns/00-base/15-typography/fonts/_fonts.scss',
+    dest: './source/_patterns/00-base/15-typography/fonts/font-sizes.json',
+    lineStartsWith: '$fs--',
+    allowVarValues: false,
+  },
+  {
+    src: './source/_patterns/00-base/15-typography/fonts/_fonts.scss',
```

Comment:
We need to make this a unique file. We really want to ensure 1 file parsed = 1 file output.